### ISSUE TITLE
Refresh Pipeline Ends after Extract Retries

### ISSUE THREAD

**Issue by bhtucker** [harrystech/arthur-redshift-etl#89]
# Summary

Using the refresh pipeline with non-zero retry attempts for extract, failed Sqoop jobs that Arthur goes on to correct will lead the 'extract' step to show a status of `FAILED`. Thus, even though data may be fully extracted, Data Pipeline will cascade the failure and `arthur.py update` will not run.

# Details

This failure definition behavior doesn't seem to be configurable in AWS. However, the cascading failure behavior is.

**Comment by bhtucker** [comment]
With a couple changes, pipelines could proceed despite failure of `extract` steps:

* *Dependency without cascading failure*: Data Pipeline dependencies control the order of execution of steps, but apparently do not necessarily cascade failures to dependencies. See https://docs.aws.amazon.com/datapipeline/latest/DeveloperGuide/dp-manage-cascade-failandrerun.html. Potentially by setting `failureAndRerunMode` to `false` or `none`, dependent steps will run after a step fails.

* *Check DynamoDB for source extraction in `update`*: Rather than rely on extract success to signal that all targeted sources are ready to load, `arthur.py update` can check for recent successful extracts. The contract of a refresh pipeline could be a 'best effort' data update (ie, refresh all selected tables that have recent extracts) or simply enforce all relations are extracted (like we do today).

**Comment by tvogels01** [comment]
Closing since a change in the refresh pipeline was implemented with PR #88 

For refresh, the check against extract events is now based on (and turned on by):
```
--scheduled-start-time #{@scheduledStartTime}
```
(Without this switch, `update` expects manifests to exist at launch time of the command.)

For rebuild, the check is enabled by:
```
--concurrent-extract
```

We should also use the start time for rebuilds and have the same command line args.  (We had some issues lately where the 15min guess turned out to cause a failure in the pipeline.)

As for the "failed" extract steps, we should specifically call this out in play books to avoid confusion.

Also, the check against extract events should be available as a standalone command to aid with triage or debugging.
